app/controllers/comment_controller.rb

```diff
@@ -30,12 +34,51 @@ def create
           end
         end
       end
-    else
+    rescue CommentError
       flash[:error] = 'The comment could not be saved.'
       render text: 'failure'
     end
   end
 
+  def create_by_token
+    @node = Node.find params[:id]
+    @user = Users.find_by_username params[:username]
+    @body = pararms[:body]
+    @token = request.headers["token"]
+
+    if @user && @user.token == @token
+      begin
+        @comment = create_comment(@node, @user, @body)
+        msg = {
```

Comment:
I get your point. I used the `raise -> rescue` workflow here because I call the `@comment.save` inside the function but wanted the calling function to know that something went wrong and do something sensible, and decided that throwing an error would be the best possible way to achieve this. If you could suggest a better way, I would be more than happy to use it instead, though.